SourceryFramework/Sources/Parsing/Composer.swift

```diff
@@ -156,44 +174,53 @@ public struct Composer {
         typeName.generic = lookupName.generic
 
         let resolvedTypeName = lookupName.generic?.name ?? lookupName.unwrappedTypeName
-        return unique[resolvedTypeName] ?? typeFromModule(resolvedTypeName, modules: modules)
+        
+        let type = unique[resolvedTypeName] ?? typeFromModule(resolvedTypeName, modules: modules)
+        if let knownPresumedType = presumedType, knownPresumedType.isConcreteGenericType {
+            let resolvedGenericTypeParameters = knownPresumedType.genericTypeParameters.map {
```

Comment:
I'd maybe move this map to `specializeType` as it's part of specialisation process anyway

Reply:
The reason I did not move this to `specializeType` is that we would need to pass `resolveTypeWithName` there, which actually makes this method to not only specialize, but resolve types too. In current form `specializeType` does one thing and one thing only - takes existing type, and creates a new one with specified generic type parameters, and does not search for underlying types.